src/history/hs.c

```diff
@@ -462,6 +464,14 @@ __hs_insert_record_with_btree_int(WT_SESSION_IMPL *session, WT_CURSOR *cursor, W
             if (cmp == 0)
                 counter = hs_counter + 1;
         }
+
+        /* Now handle the out-of-order case. */
+        if (upd->start_ts != WT_TS_NONE) {
+            WT_ERR_NOTFOUND_OK(cursor->next(cursor), true);
+            if (ret == 0)
+                WT_ERR(__hs_fixup_out_of_order_from_pos(
+                  session, cursor, btree->id, key, insert_ts, &counter));
```

Comment:
I want the thing that I'm inserting to be at the end. So I'm going to let this function bump the `counter` for every update that it fixes up and moves back to the currently inserted timestamp.